### ISSUE TITLE
Is CTCWordBeamSearch actually part of the model?

### ISSUE THREAD

**Issue by thetruejacob** [githubharald/SimpleHTR#116]
Thank you again for your continued support for this great project. I've been playing with it for the past few weeks and was wondering something regarding the importance of CTCWordbeamsearch during the training loops. As I understand it, CTCWordbeamsearch is more of a 'crutch' that boosts model performance, but does not in itself contain any parameters that are learned, and is not part of the model per se. Is this correct?

I am asking because the most time-consuming part of the training process has actually been validating performance on the validation set after each training loop, and this is because wordbeamsearch is used to calculate the character error rate/ line accuracy. If I train without wordbeamsearch at all, and use WBS only in a final inference model (in production), will I get the same performance?

**Comment by githubharald** [comment]
you're right - word beam search does not contain learnable parameters and therefore should only be used in inference.
I suggest to use best path decoding while training, it's very fast, and if results are good with best path decoding they will also be good with word beam search.
